plugins/linux-capture/xcomposite-input.cpp

```diff
@@ -0,0 +1,382 @@
+		// minimized or on offscreen workspaces or already captured on NVIDIA.
+		xcomp_create_pixmap(conn, disp, s, LOG_DEBUG);
+		xcb_xcursor_offset_win(conn, s->cursor, s->win);
+		xcb_xcursor_offset(s->cursor, s->cursor->x_org + s->crop_left,
+				   s->cursor->y_org + s->crop_top);
+	}
+
+	if (!s->tex || !s->gltex)
+		goto done;
+
+	if (xcompcap_getheight(s) == 0 || xcompcap_getwidth(s) == 0)
+		goto done;
+
+	glBindTexture(GL_TEXTURE_2D, *(GLuint *)gs_texture_get_obj(s->gltex));
+	if (s->strict_binding && s->glxpixmap) {
+		glXReleaseTexImageEXT(disp, s->glxpixmap, GLX_FRONT_EXT);
+		glXBindTexImageEXT(disp, s->glxpixmap, GLX_FRONT_EXT, nullptr);
+	}
+	gs_copy_texture_region(s->tex, 0, 0, s->gltex, s->crop_left + s->border,
+			       s->crop_top + s->border, xcompcap_getwidth(s),
+			       xcompcap_getheight(s));
+	glBindTexture(GL_TEXTURE_2D, 0);
+
+	if (s->show_cursor) {
+		xcb_xcursor_update(conn, s->cursor);
+
+		s->cursor_outside = s->cursor->x < 0 || s->cursor->y < 0 ||
+				    s->cursor->x > (int)xcompcap_getwidth(s) ||
+				    s->cursor->y > (int)xcompcap_getheight(s);
+	}
+
+done:
+	pthread_mutex_unlock(&s->lock);
+	obs_leave_graphics();
+}
+
+static void xcompcap_video_render(void *data, gs_effect_t *effect)
+{
+	gs_eparam_t *image; // Placate C++ goto rules.
+	struct xcompcap *s = (struct xcompcap *)data;
+
+	pthread_mutex_lock(&s->lock);
+
+	if (!s->tex || !s->gltex)
+		goto done;
+
+	if (s->exclude_alpha)
+		effect = obs_get_base_effect(OBS_EFFECT_OPAQUE);
+	else
+		effect = obs_get_base_effect(OBS_EFFECT_DEFAULT);
+
+	image = gs_effect_get_param_by_name(effect, "image");
+	gs_effect_set_texture(image, s->tex);
+
+	while (gs_effect_loop(effect, "Draw")) {
+		gs_draw_sprite(s->tex, 0, 0, 0);
+	}
+
+	if (s->gltex && s->show_cursor && !s->cursor_outside) {
+		effect = obs_get_base_effect(OBS_EFFECT_DEFAULT);
+
+		while (gs_effect_loop(effect, "Draw")) {
+			xcb_xcursor_render(s->cursor);
+		}
+	}
+
+done:
+	pthread_mutex_unlock(&s->lock);
+}
+
+static obs_properties_t *xcompcap_props(void *unused)
+{
+	UNUSED_PARAMETER(unused);
+
+	obs_properties_t *props = obs_properties_create();
+
+	obs_property_t *wins = obs_properties_add_list(
+		props, "capture_window", obs_module_text("Window"),
+		OBS_COMBO_TYPE_LIST, OBS_COMBO_FORMAT_STRING);
+
+	struct WindowInfo {
+		std::string lex_comparable;
+		std::string name;
+		std::string desc;
+	};
+
+	std::vector<WindowInfo> window_strings;
+	auto windows = xcomp_top_level_windows(conn);
+	for (xcb_window_t win : windows) {
+		std::string name = xcomp_window_name(conn, disp, win);
+		std::string cls = xcomp_window_class(conn, win);
+		std::string id = std::to_string((uint32_t)win);
+		std::string desc =
+			(id + WIN_STRING_DIV + name + WIN_STRING_DIV + cls);
+
+		std::string name_lower = name;
+		std::transform(name_lower.begin(), name_lower.end(),
+			       name_lower.begin(),
+			       [](unsigned char c) { return std::tolower(c); });
```

Comment:
You can replace the `std::string` usage by `dstr`, and use `dstr_to_lower()` here